### PR TITLE
fixes for issues on azure core for final fields enforcing

### PR DESCRIPTION
**Pull request by vhvb1989**

### CODE REVIEW

File: sdk/core/azure-core/src/main/java/com/azure/core/util/ExpandableStringEnum.java [hunk 1]
```diff
@@ -17,7 +17,8 @@
  * @param <T> a specific expandable enum type
  */
 public abstract class ExpandableStringEnum<T extends ExpandableStringEnum<T>> {
-    private static ConcurrentMap<String, ? extends ExpandableStringEnum<?>> valuesByName = new ConcurrentHashMap<>();
+    private static final ConcurrentMap<String,
```
Comment (conniey) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/ExpandableStringEnum.java]: I would suggest keeping the class identifier on the same line.
```java
private static ConcurrentMap<String, ? extends ExpandableStringEnum<?>> VALUES_BY_NAME =
    new ConcurrentHashMap<>();
```

File: sdk/core/azure-core/src/main/java/com/azure/core/util/configuration/Configuration.java [hunk 2]
```diff
@@ -24,13 +24,14 @@
 
     private final ClientLogger logger = new ClientLogger(Configuration.class);
 
-    private ConcurrentMap<String, String> configurations = new ConcurrentHashMap<>();
+    private final ConcurrentMap<String, String> configurations;
```
Comment (conniey) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/configuration/Configuration.java]: The instantiation could have been kept on the same line.
```java
private final ConcurrentMap<String, String> configurations = new ConcurrentHashMap<>();
```

Comment (vhvb1989) [reply sdk/core/azure-core/src/main/java/com/azure/core/util/configuration/Configuration.java]: there's a constructor that sets the value for this field
```
    private Configuration(ConcurrentMap<String, String> configurations) {
        this.configurations = new ConcurrentHashMap<>(configurations);
    }
```

and another constructor that don't
```
  public Configuration() {
      
    }
```

If we want field to be final, we need to let constructor handle assignation. 
Or do you have something else on mind that I might not considering?

Comment (conniey) [reply sdk/core/azure-core/src/main/java/com/azure/core/util/configuration/Configuration.java]: Ahh gotcha. That makese sense. I didn't see the other constructor.

**Pull request merged**